azurelinuxagent/common/event.py

```diff
@@ -406,38 +410,35 @@ def add_default_parameters_to_event(event_parameters, set_values_for_agent=True)
+        ContainerId can change due to live migration and we want to preserve the container id of the container writing
         the event, rather than sending the event.
-        OpcodeName - This is used as the actual time of event generation.
 
-        :param event_parameters: List of parameters of the event.
-        :param set_values_for_agent: Need default values populated or not. Extensions need only GAVersion and
-                                            ContainerId to be populated and others should be
-        :return: Event with default parameters populated (either values for agent or extension)
+        :param event: Event which parameters need to be expanded and updated.
+        :param is_agent_event: Whether the default values will be populated from the agent or not.
+        :return: List of final parameters to be appended to the event (whether agent or extension event).
         """
-        DefaultParameter = namedtuple('DefaultParameter', ['name', 'value'])
-        default_parameters = [DefaultParameter("GAVersion", CURRENT_AGENT),
-                              DefaultParameter('ContainerId', get_container_id_from_env()),
-                              DefaultParameter('OpcodeName', datetime.utcnow().__str__() if set_values_for_agent else ""),
-                              DefaultParameter('EventTid', threading.current_thread().ident if set_values_for_agent else 0),
-                              DefaultParameter('EventPid', os.getpid() if set_values_for_agent else 0),
-                              DefaultParameter("TaskName", threading.current_thread().getName() if set_values_for_agent else ""),
-                              DefaultParameter("KeywordName", '')]
-
         # Converting the event_parameters into a dictionary as it helps to easily look up and get values
-        param_names = OrderedDict([(param.name, param.value) for param in event_parameters])
+        event_params_dict = OrderedDict([(param.name, param.value) for param in event.parameters])
 
-        for param in default_parameters:
-            if param.name not in param_names or set_values_for_agent:
-                # If set_values_for_agent, we disregard any values already set for an existing default property and
-                # replaces it with a latest entry.
-                param_names[param.name] = param.value
-
-        parameters = DataContractList(TelemetryEventParam)
-        for name, value in param_names.items():
-            parameters.append(TelemetryEventParam(name, value))
-
-        return parameters
+        DefaultParameter = namedtuple('DefaultParameter', ['name', 'value'])
+        default_params = [DefaultParameter('GAVersion', CURRENT_AGENT),
+                          DefaultParameter('ContainerId', get_container_id_from_env()),
+                          DefaultParameter('OpcodeName', datetime.utcnow().__str__() if is_agent_event else event_params_dict.get('OpcodeName', '')),
```

Comment:
Instead of doing this could you use this - `datetime.utcnow().strftime(u'%Y-%m-%dT%H:%M:%S.%fZ')` to keep it constant with our logging.

Reply:
Updated, thanks.